tpp-app/src/home/components/DayComponent.js

```diff
@@ -0,0 +1,191 @@
+import React, { createElement } from 'react';
+import { View, SafeAreaView, StyleSheet, Text, TouchableOpacity } from 'react-native';
+import { CrampsTerribleIcon, CrampsNeutralIcon, CrampsBadIcon, CrampsGoodIcon, CrampsNoneIcon } from '../../services/utils/calendaricons';
+import { FlowHeavyIcon, FlowMediumIcon, FlowLightIcon, FlowNoneIcon, FlowSpottingIcon } from '../../services/utils/calendaricons';
+import { MoodHappyIcon, MoodSadIcon, MoodNeutralIcon, MoodSickIcon, MoodAngryIcon, MoodLolIcon, MoodIdkIcon, MoodGreatIcon, MoodLovedIcon } from '../../services/utils/calendaricons';
+import { ExerciseBallSportIcon, ExerciseCardioIcon, ExerciseCycleSportIcon, ExerciseMartialArtsIcon, ExerciseRacketSportsIcon, ExerciseStrengthIcon, ExerciseWaterSportIcon, ExerciseWinterSportIcon, ExerciseYogaIcon} from '../../services/utils/calendaricons';
+import { VIEWS } from '../../services/utils/constants';
+import { FILTER_COLOURS, FILTER_TEXT_COLOURS } from '../../services/utils/constants';
+import { STACK_SCREENS } from '../CalendarNavigator';
+
+// The component that is used by each day in the calendar
+export const DayComponent = ({ date, state, marking, selectedView, navigation }) => {
+
+    let bgColor;
+    let textColor;
+    let iconName = 'view';
+    let renderedIcon;
+
+    // If this specific date has been marked
+    if (marking) {
+        // View key just tells us what the view is set to
+        let viewKey = getKeyByValue(VIEWS, selectedView).toLowerCase()
+        // Basically whatever special value that is attached to the specific key in the Symptoms object
+        // i.e. for flow it would be HEAVY/MEDIUM/LIGHT
+        // for sleep it will be a number etc.
+        let symptomAttribute = marking.symptoms[viewKey]
+
+        // If it contains a working attribute
+        if (symptomAttribute) {
+            
+            let attribute = symptomAttribute
+            
+            // Sleep and exercise have special cases to be calculated through
+            switch (viewKey) {
+                case 'sleep':
+                    attribute = filterSleep(attribute)
+                    break;
+                case 'exercise':
+                    attribute = filterExercise(symptomAttribute.exercise_minutes)
+                    iconName = viewKey + symptomAttribute.exercise.toLowerCase()
+                    iconName = iconName.replace(/\s+/g, '')
+
+                    break;
+            }
+
+            
+            // Mood is the only one that does not modify the background colour
+            if (viewKey !== 'mood') {
+                bgColor = FILTER_COLOURS[viewKey.toUpperCase()][attribute.toUpperCase()]
+                textColor = FILTER_TEXT_COLOURS[viewKey.toUpperCase()][attribute.toUpperCase()]
+            }
+            
+            // Get Icon
+            if (viewKey !== 'sleep' && viewKey !== 'exercise') {
```

Comment:
nitpicky but u could just put this in an if-else with the switch statement from line 34 (decomposed into if statements). Not that deep tho